app/src/main/java/com/example/trainingintive/util/Events.kt

```diff
@@ -10,5 +12,9 @@ sealed class SplashScreenEvent : Event {
 }
 
 sealed class MainScreenEvent : Event {
+    class Error(exception: Exception) : Event {
+        val messageId = ErrorMessageId.getId(exception)
```

Comment:
You should avoid any additional logic in navigators/events. They should be primitive like – "Hey, navigate to screen X", "Hey, show message Y in snackbar", etc. Extract messageId in ViewModel and pass it to Navigator.